server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -903,6 +906,24 @@ public String toString() {
         return "IngestDocument{" + " sourceAndMetadata=" + ctxMap + ", ingestMetadata=" + ingestMetadata + '}';
     }
 
+    public void redirect(String destIndex) {
+        getMetadata().setIndex(destIndex);
+        invokeDefaultPipelineOfDestination = true;
+        skipCurrentPipeline = true;
+    }
+
+    boolean isInvokeDefaultPipelineOfDestination() {
+        return invokeDefaultPipelineOfDestination;
+    }
+
+    boolean isSkipCurrentPipeline() {
+        return skipCurrentPipeline;
+    }
+
+    void resetPipelineSkipping() {
+        skipCurrentPipeline = false;
+    }
```

Comment:
Why doesn't this also reset `invokeDefaultPipelineOfDestination`? Also, it's unclear from the code why we need this method to be invoked.

Reply:
Good point, it needs to reset both. I've changed it to just be a single boolean.